lib/engine/game/g_18_india/game.rb

```diff
@@ -913,6 +915,53 @@ def companies_to_payout(ignore: [])
           end
         end
 
+        # ----- Route Modificatons for Gauge Change stops (modifed from 1848)
+
+        # Add gauge changes to visited stops, they count as 0 revenue City stops,
+        def visited_stops(route)
+          gauge_changes = edge_crossings(route)
+          route_stops = route.connection_data.flat_map { |c| [c[:left], c[:right]] }.uniq.compact # super
+          return route_stops unless gauge_changes.positive?
+
+          LOGGER.debug "GAME::visited_stops > gauge_changes: #{gauge_changes} route_stops: #{route_stops}"
+          add_gauge_changes_to_stops(gauge_changes, route_stops)
+        end
+
+        def add_gauge_changes_to_stops(num, route_stops)
+          return [] unless num.positive?
+
+          gauge_changes = Array.new(num) { Engine::Part::City.new('0') }
+          first_stop = route_stops.first
+          gauge_changes.each do |stop|
+            stop.tile = first_stop.tile
```

Comment:
It might be worth creating a GaugeChangeStopclass that's a subclass of Engine::Part::City that defaults to zero revenue.
If nothing else, it would let you rewrite this method as a single line:
```
route_stops.insert(1, *Array.new(num) { GaugeChangeStop.new })
```

Reply:
> What's this line for? Does this information end up being shown to the user in the route selector? It's going to be confusing if you have the same hex coordinates repeated several times.

The Hex isn't shown as part of the "route info" when selecting routes.  However, I needed it as there was another method that broke when the visited hexes had a `nil` value.

Reply:
changing the class of the stop breaks other route checks.

Reply:
![image](https://github.com/tobymao/18xx/assets/118568589/80f5d13a-783c-445b-bff9-8edfcb304672)